macos/Classes/FilePickerPlugin.swift

```diff
@@ -0,0 +1,123 @@
+import Cocoa
+import FlutterMacOS
+
+public class FilePickerPlugin: NSObject, FlutterPlugin {
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let channel = FlutterMethodChannel(
+      name: "miguelruivo.flutter.plugins.filepicker",
+      binaryMessenger: registrar.messenger)
+    let instance = FilePickerPlugin()
+    registrar.addMethodCallDelegate(instance, channel: channel)
+  }
+
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult)
+  {
+    switch call.method {
+    case "pickFiles":
+      handleFileSelection(call, result: result)
+
+    case "getDirectoryPath":
+      handleDirectorySelection(call, result: result)
+
+    case "saveFile":
+      handleSaveFile(call, result: result)
+
+    default:
+      result(FlutterMethodNotImplemented)
+    }
+  }
+
+  private func handleFileSelection(
+    _ call: FlutterMethodCall, result: @escaping FlutterResult
+  ) {
+    let dialog = NSOpenPanel()
+    let args = call.arguments as! [String: Any]
+
+    dialog.showsResizeIndicator = true
+    dialog.directoryURL = URL(
```

Comment:
What if the directory url is invalid? Do we need to handle this with a `result(FlutterError())` ? Or do we not set the url?

Reply:
macOS falls back to the default or previous correct folder. I can add this to doc

Reply:
updated docs